src/InlineRelationsTransformer.cpp

```diff
@@ -0,0 +1,993 @@
+#include "AstTransforms.h"
+#include "AstVisitor.h"
+
+namespace souffle {
+
+template <class T>
+class NullableVector {
+private:
+    std::vector<T> vector;
+    bool valid;
+
+public:
+    NullableVector() : valid(false) {}
+    NullableVector(std::vector<T> vector) : vector(vector), valid(true) {}
+
+    bool isValid() const {
+        return valid;
+    }
+
+    const std::vector<T>& getVector() const {
+        assert(valid && "Accessing invalid vector!");
+        return vector;
+    }
+};
+
+NullableVector<std::vector<AstLiteral*>> getInlinedLiteral(AstProgram&, AstLiteral*);
+
+/**
+ * Replace constants in the head of inlined clauses with (constrained) variables.
+ */
+void normaliseInlinedHeads(AstProgram& program) {
+    static int newVarCount = 0;
+
+    // Go through the clauses of all inlined relations
+    for (AstRelation* rel : program.getRelations()) {
+        if (!rel->isInline()) {
+            break;
+        }
+
+        for (AstClause* clause : rel->getClauses()) {
+            // Set up the new clause with an empty body and no arguments in the head
+            auto newClause = std::make_unique<AstClause>();
+            newClause->setSrcLoc(clause->getSrcLoc());
+            auto clauseHead = std::make_unique<AstAtom>(clause->getHead()->getName());
+            newClause->setHead(std::move(clauseHead));
```

Comment:
Sooo, now that clauseHead is a unique pointer and its contents have been moved into newClause, it can't be used. newclause->getHead() works, though.